### PR TITLE
Add Prettier

### PR DESCRIPTION
**Pull request by Apollon77**
first only for backend files, please @foxriver76 check the rules and apply for Frontend in extra PR

### CODE REVIEW

File: .github/workflows/test-and-release.yml [hunk 1]
```diff
@@ -55,7 +55,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node-version: [18.x, 20.x, 22.x]
+        node-version: [18.x, 20.x, 22.4.1]
```
Comment (foxriver76) [inline .github/workflows/test-and-release.yml:58]: Is there a reason to be that precise?^

File: .eslintrc.js [hunk 2]
```diff
@@ -13,20 +13,10 @@ module.exports = {
         'plugin:@typescript-eslint/recommended', // Uses the recommended rules from the @typescript-eslint/eslint-plugin
```
Comment (foxriver76) [inline .eslintrc.js:13]: Should we maybe use someting like https://github.com/ioBroker/ioBroker.js-controller/blob/f54793e0236711426849de067a94a1dcf426fbfe/.eslintrc.json#L8 

https://github.com/ioBroker/ioBroker.js-controller/blob/f54793e0236711426849de067a94a1dcf426fbfe/package.json#L60
https://github.com/ioBroker/ioBroker.js-controller/blob/f54793e0236711426849de067a94a1dcf426fbfe/package.json#L62

**Review by foxriver76 — APPROVED** [review]

File: .github/workflows/test-and-release.yml (hunk 1 above)
Comment (Apollon77) [reply .github/workflows/test-and-release.yml:58]: yes a current npm bug ... else you get https://github.com/ioBroker/ioBroker.matter/actions/runs/10021413829/job/27700078813#step:4:16 ... maybe we can revert when npm fixed that

File: .eslintrc.js (hunk 2 above)
Comment (Apollon77) [reply .eslintrc.js:13]: sure

**Pull request merged**